src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts

```diff
@@ -410,10 +410,14 @@ export class ConsultationQuestionnaireComponent implements OnInit, AfterViewInit
   }
 
   getConsultationResponse() {
+    const typeResponse = {
+      true: 'shared',
+      false: 'anonymous'
+    };
     const consultationResponse =  {
       consultationId: this.profileData.id,
       satisfactionRating : this.responseFeedback,
-      visibility: this.responseVisibility ? 'shared' : 'anonymous',
+      visibility: this.currentUser ? typeResponse[this.responseVisibility && this.currentUser.isVerified] : 'anonymous',
```

Comment:
This is very confusing, it can be rewritten as `this.responseVisibility && this.currentUser?.isVerfified ? "shared" : "anonymous"`